abel/tools.py

```diff
@@ -7,89 +7,196 @@
 
 import numpy as np
 from scipy.ndimage import map_coordinates
+from scipy.ndimage.interpolation import shift
+from scipy.optimize import curve_fit, minimize 
 
-def calculate_speeds(IM,origin=None):
-    """ This performs an angular integration of the image and returns the one-dimentional intensity profile 
-        as a function of the radial coordinate. It assumes that the image is properly centered. 
+def calculate_speeds(IM, origin=None, Jacobian=False, dr=1, dt=None):
+    """ This performs an angular integration of the image returning 
+        the one-dimentional intensity profile as a function of the 
+        radial coordinate. 
         
      Parameters
      ----------
-      - IM: a NxN ndarray.
+      - IM: a rows x cols ndarray.
+      - origin: tuple, image center coordinate 
+                defaults to (rows//2+rows%2,cols//2+cols%2)
+      - Jacobian: boolean, include sinθ in the angular sum (integration)
+      - dr: float, radial coordinate grid spacing, in pixels (default 1)
+      - dt: float, theta coordinate grid spacing in degrees
+            (default rows//2)
       
      Returns
      -------
-      - speeds: a 1D array of the integrated intensity versus the radial coordinate.
+      - speeds: a 1D array of the integrated intensity 
+      - r: the 1D array of radial coordinates.
+      - theta: 1D array of theta coordinates
      """
     
-    polarIM, ri, thetai = reproject_image_into_polar(IM,origin)
+    polarIM, r, theta = reproject_image_into_polar(IM, origin, rmax='circle', dr=dr, dt=dt)
 
+    if Jacobian:  #  x r^2 sinθ    
+        sintheta = np.abs(np.sin(theta))
+        polarIM = polarIM*sintheta[np.newaxis, :]
+        r2 = r**2
+        polarIM = polarIM*r2[:,np.newaxis]
+    
     speeds = np.sum(polarIM, axis=1)
     
-    # Clip the data corresponding to the corners, since these pixels contain incomplete information
+    # Clip the data corresponding to the corners, since 
```

Comment:
Since you are using ``rmax='circle``, then you should not need to crop the image, right?